server/libs/storage/src/interfaces/immich-storage.interface.ts

```diff
@@ -0,0 +1,4 @@
+export interface IImmichStorage {
```

Comment:
This will probably need a few more functions, like deleting, checking whether a file exists, and maybe getting metadata (eg size).

Reply:
And all the functions in this interface should take some object with asset metadata that they can use to find the file in their backend.
Some potential fields:
* `assetId`
* `userId`
* `assetType`
* `originalFileName`
* `createdAt`
* `checksum`

Reply:
It doesn't look like we're using this interface in this PR, can we remove it and related artifacts?